Stop notification when App is at End of life

AndroidMedaGalaxy opened the pull request:
* Stopping background worker when user is tested positive and they submit the keys.

<!--
Thank you for supporting us with your Pull Request! 🙌 ❤️
Before submitting, please take the time to check the points below and provide some descriptive information.
-->

## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [JIRA Issue 1792: Android - Notifications although app is in end-of-life](https://jira.itc.sap.com/browse/EXPOSUREAPP-1792)
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [x] Make sure that your PR does not contain changes in text files, therefore the PR must not contain changes in values/strings/* and / or assets/* (see issue #332 for further information)
* [x] Make sure that your PR does not contain compiled sources (already set by the default .gitignore) and / or binary files

## Description
* When user tests positive, they can submit the keys.
* Upon successful keys submission, app comes to end of life.
* Hence the background jobs have to be stopped to prevent unnecessary notifications

## Change log
* Condition to check a successful key submission is added.
* Upon success, BackgroundWorkScheduler is stopeed

AndroidMedaGalaxy committed:
Stop notification when at End of life

* Stopping background worker when user is tested positive and they submit the keys.

pwoessner committed:
Merge branch 'dev' into fix/app-notification-stop

pwoessner committed:
Merge branch 'dev' into fix/app-notification-stop

AlexanderAlferov reviewed (changes requested):
We need to double check

AndroidMedaGalaxy committed:
Stop background services when diagnosis keys are submitted

AndroidMedaGalaxy committed:
check added before starting backgrouns schedular

To prevent Backgrond services from starting again, a check has been added that checks if diagnosis keys are submitted already.

pwoessner committed:
Merge branch 'dev' into fix/app-notification-stop

AndroidMedaGalaxy committed:
Merge branch 'dev' into fix/app-notification-stop

AndroidMedaGalaxy committed:
Merge branch 'fix/app-notification-stop' of https://github.com/corona-warn-app/cwa-app-android into fix/app-notification-stop

AndroidMedaGalaxy committed:
tests modified based on code changes
